### DISCUSSION TITLE
Is it intentional that `Field(None)` leads to typing errors but `Field(default=None)` does not?

### DISCUSSION THREAD

**Discussion by smorokin** [pydantic/pydantic#9413] [Question] [answered]
Sorry, I could not really find an answer - just that something very similar was fixed (e.g. #3617)

Example Code:
```python
from pydantic import BaseModel, Field


class User1(BaseModel):
    name: str | None = Field(None)


class User2(BaseModel):
    name: str | None = Field(default=None)


u1 = User1() # mypy & pyright error: Missing named argument "name" for "User1"  [call-arg]

u2 = User2() # no typing error
```

The example works either way, but both pyright and mypy report a typing error. Is this by design?

**Comment by Viicos** [comment, marked as answer]
Pydantic relies on [`PEP 681`](https://peps.python.org/pep-0681/#field-specifier-parameters) to make type checkers aware of the signature of models. The PEP [requires](https://peps.python.org/pep-0681/#field-specifier-parameters) `Field` arguments to be keyword only, but `default` can be set as a positional argument for historical reasons (it was here before PEP 681 was a thing, and was kept as is for compatibility reasons).

It is thus recommended to use the explicit `default` keyword argument, or, as an alternative, `name: str | None = None` (or using `Annotated`, see https://github.com/pydantic/pydantic/discussions/7379#discussioncomment-6951557).

**Reply by smorokin** [reply]
Thank you, that makes it so much clearer!
